hvac_services/demo/purchase_order_post.xml

```diff
@@ -0,0 +1,118 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<odoo noupdate="1">
+    <function model="purchase.order" name="button_confirm">
+        <value eval="[
+            ref('purchase_order_1'),
+            ref('purchase_order_2')]"/>
+    </function>
+    <function name="write" model="stock.move.line">
+        <value model="stock.move.line" eval="obj().search([
+        ('product_id', '=', ref('product_product_6')),
+        ('picking_id', '=', obj().env.ref('hvac_services.purchase_order_2').picking_ids.id)])[0].id"/>
```

Comment:
To avoid the slice, you could limit the search to 1, and add a criterion that lot_id is False in the domain (same as in it_hardware for reference)